slack/web/slack_response.py

```diff
@@ -132,13 +135,27 @@ def __next__(self):
                 {"cursor": self.data["response_metadata"]["next_cursor"]}
             )
 
-            response = asyncio.get_event_loop().run_until_complete(
-                self._client._request(
-                    http_verb=self.http_verb,
-                    api_url=self.api_url,
-                    req_args=self.req_args,
+            if self._use_sync_aiohttp:
+                # We no longer recommend going with this way
+                response = asyncio.get_event_loop().run_until_complete(
+                    self._client._request(
+                        http_verb=self.http_verb,
+                        api_url=self.api_url,
+                        req_args=self.req_args,
+                    )
                 )
-            )
+            else:
+                if self._client is None:
+                    # NOTE: It's not possible to support this due to circular import issues.
+                    msg = (
+                        "Directly using UrllibWebClient doesn't support response pagination as iteration. "
+                        "Use WebClient with run_async=False and use_sync_aiohttp=False."
+                    )
+                    raise e.SlackRequestError(msg)
+                response = self._client._sync_request(
```

Comment:
As this method is not a coroutine, using sync client also for `run_async=True` clients. Regarding `run_async=True`, it works anyways but we can revisit this to make it completely non-blocking in the future.

Reply:
The changes here fixes #626